src/ILVerify/tests/ILMethodTester.cs

```diff
@@ -0,0 +1,208 @@
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using Internal.IL;
+using Internal.TypeSystem.Ecma;
+using Xunit;
+
+namespace ILVerify.Tests
+{
+    abstract class TestCase
+    {
+        public string MethodName { get; set; }
+        public ILImporter ILImporter { get; set; }
+    }
+
+    /// <summary>
+    /// Describes a test case with a method that contains valid IL
+    /// </summary>
+    class ValidILTestCase : TestCase { }
+
+    /// <summary>
+    /// Describes a test case with a method that contains invalid IL with the expected VerifierErrors
+    /// </summary>
+    class InvalidILTestCase : TestCase
+    {
+        public List<VerifierError> ExpectedVerifierError { get; set; }
+
+        /// <summary>
+        /// The expected exception during validation.
+        /// If no exception is thrown during verification this must be null.
+        /// </summary>
+        public bool ExpectsException { get; set; }
+    }
+
+    /// <summary>
+    /// Parses the methods in the test assemblies. 
+    /// It loads all assemblies from the test folder defined in <code>TestDataLoader.TESTASSEMBLYPATH</code>
+    /// This class feeds the xunit Theories
+    /// </summary>
+    class ILReader
+    {
+        public enum TestMethodType
+        {
+            Valid, Invalid
+        }
+
+        /// <summary>
+        /// Returns all methods that contain valid IL code based on the following naming convention:
+        /// [FriendlyName]_Valid
+        /// The method must contain 1 '_'. The part before the '_' is a friendly name describing what the method does. 
+        /// The word after the '_' has to be 'Valid'. 
+        /// E.g.: 'SimpleAdd_Valid'
+        /// </summary>
+        public static IEnumerable<Object[]> GetMethodsWithValidIL()
+        {
+            var methodSelector = new Func<string[], EcmaMethod, EcmaMethodIL, TestCase>((mparams, method, methodIL) =>
+            {
+                if (mparams.Length == 2 && mparams[1].ToLower() == "valid")
+                {
+                    return new ValidILTestCase { ILImporter = new ILImporter(method, methodIL) };
+                }
+                return null;
+            });
+            return GetTestMethodsFromDll(methodSelector);
+        }
+
+        /// <summary>
+        /// Returns all methods that contain valid IL code based on the following naming convention:
+        /// [FriendlyName]_Invalid_[ExpectedVerifierError1].[ExpectedVerifierError2]_[ExpectedExceptionDuringValidation]
+        /// The method name must contain 4 '_' characters.
+        /// 1. part: a friendly name
+        /// 2. part: must be the word 'Invalid'
+        /// 3. part: the expected VerifierErrors as string separated by '.'.
+        /// 4. part: Either 'ExpectsException' or nothing. If 'ExpectsException' than ILVerify is expected to throw an exception during verification, otherwise not
```

Comment:
>  they will be always caught internally

That sounds good, but the current implementation crashes with exceptions in many cases (e.g. with my first simple add test) and we have to handle this somehow in the test code in order to be able to add and run tests as soon as possible. 

My suggestion: I remove this 4. part (so we ignore this in the name of the test methods) and I simply add an empty catch block in the test code to make sure the test method does not fail when ILVerify crashes. Once the 
`I would expect that internal verifier exceptions would not be part of the public surface - they will be always caught internally.` is implemented we can remove the empty catch block